api/repositories/process_repository_test.go

```diff
@@ -4,131 +4,142 @@ import (
 	"context"
 	"time"
 
-	. "github.com/onsi/gomega/gstruct"
-
 	. "code.cloudfoundry.org/cf-k8s-controllers/api/repositories"
 	workloadsv1alpha1 "code.cloudfoundry.org/cf-k8s-controllers/controllers/apis/workloads/v1alpha1"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
 	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
 )
 
 var _ = Describe("ProcessRepo", func() {
 	var (
-		testCtx      context.Context
-		processRepo  *ProcessRepo
-		namespace    *corev1.Namespace
-		app1GUID     string
-		app2GUID     string
-		process1GUID string
-		process2GUID string
+		testCtx        context.Context
+		processRepo    *ProcessRepo
+		org            *v1alpha2.SubnamespaceAnchor
+		space1, space2 *v1alpha2.SubnamespaceAnchor
+		app1GUID       string
+		app2GUID       string
+		process1GUID   string
+		process2GUID   string
+
+		cfApp1 *workloadsv1alpha1.CFApp
+		cfApp2 *workloadsv1alpha1.CFApp
+
+		cfProcess1 *workloadsv1alpha1.CFProcess
+		cfProcess2 *workloadsv1alpha1.CFProcess
+
+		spaceDeveloperClusterRole *rbacv1.ClusterRole
 	)
 
 	BeforeEach(func() {
 		testCtx = context.Background()
 
-		processRepo = NewProcessRepo(k8sClient)
+		processRepo = NewProcessRepo(k8sClient, userClientFactory)
 
-		namespaceName := generateGUID()
-		namespace = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
-		Expect(
-			k8sClient.Create(testCtx, namespace),
-		).To(Succeed())
+		rootNs := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: rootNamespace}}
+		Expect(k8sClient.Create(testCtx, rootNs)).To(Succeed())
+
+		org = createOrgAnchorAndNamespace(testCtx, rootNamespace, prefixedGUID("org"))
+		space1 = createSpaceAnchorAndNamespace(testCtx, org.Name, prefixedGUID("space1"))
+		space2 = createSpaceAnchorAndNamespace(testCtx, org.Name, prefixedGUID("space2"))
+
+		app1GUID = prefixedGUID("app1")
+		app2GUID = prefixedGUID("app2")
+		process1GUID = prefixedGUID("process1")
+		process2GUID = prefixedGUID("process2")
 
-		app1GUID = generateGUID()
-		app2GUID = generateGUID()
-		process1GUID = generateGUID()
-		process2GUID = generateGUID()
+		spaceDeveloperClusterRole = createClusterRole(testCtx, SpaceDeveloperClusterRoleRules)
 	})
 
 	AfterEach(func() {
 		Expect(
```

Comment:
I believe that for env tests cleaning up the namespaces makes no sense, as the test env gets destroyed anyway. Also, as in env tests there are no controllers, deleting the namespace does not delete the objects it contains, they remain as records in the etcd.
TL;DR I don't think this `AfterEach` is needed at all.